Extract lustre-ldisk-zfs + add service file

jgrund opened the pull request:
Extract the lustre-ldisk-zfs spec + add a
service file that ensures ZFS services are
initialized (and kernel module is loaded) during & after first install.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
git subrepo clone (merge) /Users/joegrund/repos/intel-hpdd/module-tools ./include

subrepo:
  subdir:   "include"
  merged:   "53ff444"
upstream:
  origin:   "/Users/joegrund/repos/intel-hpdd/module-tools"
  branch:   "master"
  commit:   "53ff444"
git-subrepo:
  version:  "0.4.0"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "74339e8"

Commit:
Extract lustre-ldisk-zfs + add service file

Extrace the lustre-ldisk-zfs spec + add a
service file that ensures ZFS services are
initialized (and kernel module is loaded) during & after first install.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
updates

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
fixup

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
remove user bit

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
update makefile

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
full version

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add Source0

Signed-off-by: Joe Grund <joe.grund@intel.com>

brianjmurrell commented on .travis.yml:
```diff
@@ -0,0 +1,31 @@
+    before_install:
+    - docker pull centos:centos7
+    script:
+    - ./include/travis/run_in_centos7_docker.sh include/travis/mock_build_test.sh
```
You should add an _rpmlint_ test job here that runs `make rpmlint`.

brianjmurrell commented on Makefile:
```diff
@@ -0,0 +1,6 @@
+DIST_VERSION    := $(PACKAGE_VERSION)
+PACKAGE_RELEASE := 4
+
+include ./include/rpm.mk
```
So ultimately, this Makefile should probably build the tarball you need for building an RPM.

Commit:
add discrete make target for tgz

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add rpmlint check

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add quotes

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
more updates

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
Move file into place during install.

brianjmurrell commented on .travis.yml:
```diff
@@ -14,6 +14,7 @@ jobs:
     before_install:
     - docker pull centos:centos7
     script:
+    - ./include/travis/run_in_centos7_docker.sh make rpmlint
```
I thinik you should be able to chain these together to run in the same docker instance:

```
- ./include/travis/run_in_centos7_docker.sh make rpmlint && ./include/travis/run_in_centos7_docker.sh include/travis/mock_build_test.sh
```

Pity that the native linux releaes in Travis is only the Trusty release of Ubuntu.  _rpmlint_ was added in _Xenial_, which is the next LTS after _Trusty_.

jgrund replied:
This won't work, as `make` isn't installed in the container.

We should probably move some of the packages installed in ```include/travis/mock_build_test.sh``` into ```./include/travis/run_in_centos7_docker.sh``` so we have a basic environment to run things like rpmlint.

brianjmurrell replied:
So with https://github.com/intel-hpdd/lustre-ldiskfs-zfs/pull/1/commits/d099bac0d2c93519ecf6e6d5f9adbb5cb594af65, you could now chain these together to run in the same container, yes?
Or even just add the `make rpmlint` to the `mock_build_test.sh`.

jgrund replied:
Yes. I think adding make rpmlint to the test is the way to go.

brianjmurrell committed:
%prep needs %setup to unpack tarball

Commit:
move install of basic components into run in docker script.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add in rpmlint as dep

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add verbose flag to mock

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
enable and start unit post install.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
start zed

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
explicitly start service + zed. Zed doesn't have right dependencies in 0.7.x

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
add comment

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
Add another Before since zfs-zed doesn't depend on zfs.target
like it should.

brianjmurrell commented:
Keep in mind, all of the changes in `include/` need to get pushed to _module-tools_ (also).

Personally, I prefer to add those changes to _module-tools_ first and then pull them into the project that needs them.  But that is just my personal preference.  There is no issue with adding them here, then pushing them up to _module-tools_ after the fact.

Commit:
Namespace unit in case it get's pulled into ZFS.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
move rpmlint into script

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
fixup

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
revert dependency install location.

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
another fixup

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
install rpmlint

Signed-off-by: Joe Grund <joe.grund@intel.com>

brianjmurrell commented on lustre-ldiskfs-zfs.spec:
```diff
@@ -1,7 +1,7 @@
 BuildRequires: systemd
 
-%define	unit_name zfs-import-none.service
+%define	unit_name iml-zfs-import-none.service
```
Good thinking!  Would be nice to know what upstream are thinking about doing about this problem if anything.

Commit:
remove old subrepo

Signed-off-by: Joe Grund <joe.grund@intel.com>

Commit:
git subrepo clone git@github.com:intel-hpdd/module-tools.git ./include/

subrepo:
  subdir:   "include"
  merged:   "80d034f"
upstream:
  origin:   "git@github.com:intel-hpdd/module-tools.git"
  branch:   "master"
  commit:   "80d034f"
git-subrepo:
  version:  "0.4.0"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "74339e8"

brianjmurrell commented on include/travis/mock_build_test.sh:
```diff
@@ -0,0 +1,37 @@
+if ! su - mocker <<EOF; then
+set -xe
+cd "$MAPPED_DIR"
+make rpmlint
```
It would be nicer if this were a top-level test in it's own stage, rather than being overloaded into another test.  I wonder if it's worth an effort for us to [back]port some stuff to Trusty (thinking _rpmlint_ and _mock_ for starters) so that we can do morem natively, in Travis' Trusty Linux environment.

Do we have any idea how long Travis will continue with Trusty, or at least not provide an option to use Xenial, where more of these tools are available?

jgrund replied:
> Do we have any idea how long Travis will continue with Trusty, or at least not provide an option to use Xenial, where more of these tools are available?

I looked the other day, they said Xenial was right around the corner here: https://blog.travis-ci.com/2017-12-01-new-update-schedule-for-linux-build-images

brianjmurrell commented on lustre-ldiskfs-zfs.spec:
```diff
@@ -0,0 +1,55 @@
+
+%define	unit_name iml-zfs-import-none.service
+
+Name:	lustre-ldiskfs-zfs
```
If you find a need to push anything more, lining this one up with everything below would be visually pleasing.

jgrund replied:
done

brianjmurrell commented on lustre-ldiskfs-zfs.spec:
```diff
@@ -0,0 +1,55 @@
+Summary:	Package to install a Lustre storage server with both ldiskfs and ZFS support
+
+License:	MIT
+URL:	https://github.com/intel-hpdd/%{name}
```
Ditto.

jgrund replied:
done

brianjmurrell commented on lustre-ldiskfs-zfs.spec:
```diff
@@ -0,0 +1,55 @@
+%define	unit_name iml-zfs-import-none.service
+
+Name:	lustre-ldiskfs-zfs
+Version:	1
```
Is this really still version 1 or should it be 2 now that there is actually a source package (rpm versions are updated when the source they are packaging is updated) being included?

jgrund replied:
Bumped

Commit:
CR fixups.

Signed-off-by: Joe Grund <joe.grund@intel.com>
